packages/components/data-resource/data-resource-create-form.tsx

```diff
@@ -141,13 +139,21 @@ const DataResourceCreateForm: React.FC<DataResourceCreateFormProps> =
         errorMessage,
       } = props;
       const { provider } = formDataState;
+      const handleNsNameTextInputChange = (strVal: string) => {
+        formDataDispatch({ type: createFormAction.SET_NAME, value: strVal });
+        if (formDataState.secretName)
+          formDataDispatch({
+            type: createFormAction.SET_SECRET_NAME,
+            value: '',
+          });
+      };
```

Comment:
It will work with "Switch to credentials" option... 
but when "Switch to Secrets" is selected, let's say I selected Secret from the dropdown then move ahead and type resource name, it will reset the Secret that I selected from the dropdown.